__test__/generateWallets.ts

```diff
@@ -1,8 +1,47 @@
+import { setBalance } from '@nomicfoundation/hardhat-toolbox-viem/network-helpers';
 import { viem } from 'hardhat';
+import type { Address } from 'viem';
 import { publicActions } from 'viem';
+import { privateKeyToAccount, generatePrivateKey } from 'viem/accounts';
 
-export async function generateWallets() {
-  const [adminAccount, userAccount, secondUserAccount, thirdUserAccount] = await viem.getWalletClients();
+export type GeneratedWalletConfig = {
+  initialEthBalance?: number;
+};
+
+export async function generateWallets({ initialEthBalance = 1 }: GeneratedWalletConfig = { initialEthBalance: 1 }) {
```

Comment:
I discovered we weren't initialising contracts with wallets, but the default wallet with viem, which is why it automagically worked.

this fixes generate wallets so we for sure always get different wallets